components/brave_sync/brave_sync_prefs.cc

```diff
@@ -102,18 +102,31 @@ std::string Prefs::GetSeedPath() {
   return kSyncV2Seed;
 }
 
-std::string Prefs::GetSeed() const {
+std::string Prefs::GetSeed(bool* failed_to_decrypt) const {
+  if (failed_to_decrypt)
```

Comment:
Now that every caller passes a `failed_to_decrypt` bool, perhaps we can assume it's there and add a `CHECK(failed_to_decrypt)` to make sure it's not a NULL pointer?

Reply:
Sounds good, thanks.
This is fixed in https://github.com/brave/brave-core/pull/11935/commits/06eb335e68ad44357ad7410e11137f5022636553 , plus unit test to ensure `CHECK` is triggered, thanks to @darkdh for sharing this knowledge.

Also PR was re-based to the latest `master`.